### PR TITLE
ICU-22874 Update ICU4C to use a current version of the autotools

### PR DESCRIPTION
**Pull request by roubert**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22874
- [x] Required: The PR title must be prefixed with a JIRA Issue number.
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number.
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: icu4c/source/configure.ac [hunk 1]
```diff
@@ -6,15 +6,16 @@ AC_COPYRIGHT([ Copyright (C) 2016 and later: Unicode, Inc. and others. License &
...
+AC_PREREQ([2.72])
 
 # Process this file with autoreconf to produce a configure script
 AC_INIT([ICU],
-        m4_esyscmd_s([sed -n 's/^[ 	]*#[ 	]*define[ 	]*U_ICU_VERSION[ 	]*"\([^"]*\)".*/\1/p' "./common/unicode/uvernum.h]"),
-        [http://icu-project.org/bugs],
-        [International Components for Unicode],
-        [http://icu-project.org])
+        m4_esyscmd_s([sed -n 's/^[ 	]*#[ 	]*define[ 	]*U_ICU_VERSION[ 	]*"\([^"]*\)".*/\1/p' common/unicode/uvernum.h]),
+        [https://icu.unicode.org/bugs],
```
Comment (srl295) [inline icu4c/source/configure.ac:16]: it must be 2024!

**Review by srl295 — APPROVED** [review]

File: icu4c/source/configure.ac (hunk 1 above)
Comment (roubert) [reply icu4c/source/configure.ac:16]: 🤣 Updating the URLs wasn't strictly necessary to make autoconf 2.72 accept this without warnings, but I just couldn't resist sneaking that change in …

**Comment by markusicu** [comment]
Thank you!
@roubert or @srl295 do we need to update the relevant LICENSE file sections for this new version?
https://github.com/unicode-org/icu/blob/main/LICENSE#L464-L525

@yumaoka FYI

**Comment by roubert** [comment]
> @roubert or @srl295 do we need to update the relevant LICENSE file sections for this new version?

I can't imagine why that would be necessary (but on the other hand, I'm generally not very good at imagining such stuff).

**Comment by markusicu** [comment]
> > @roubert or @srl295 do we need to update the relevant LICENSE file sections for this new version?
> 
> I can't imagine why that would be necessary (but on the other hand, I'm generally not very good at imagining such stuff).

Well, the license texts should be in sync between files that have special license and our collective LICENSE file.

I just checked, and only one character changed in the license text in one of these files
- https://github.com/unicode-org/icu/pull/3182

**Comment by roubert** [comment]
> I just checked, and only one character changed in the license text in one of these files
> 
> * [ICU-22874 sync config.guess license #3182](https://github.com/unicode-org/icu/pull/3182)

OK, changing that semicolon to a comma is definitely something that I could never have imagined that it would matter,

**Comment by markusicu** [comment]
It probably doesn't matter, but it's good to be in sync.

**Pull request merged**